### PR TITLE
Improve database connection

### LINKED ISSUE #11: Improve database connection
#### Why this is important

Currently, the database connection is tied to a single cluster url, hence, it will not be convenient to establish connection from a different development source.

As a collaborator, I should be able to provide database credentials to ease my development workflow.

### PR DESCRIPTION
**Pull request by devdbrandy**
Provides minimal enhancement to manner in which database connection is handled. This closes #11 

- Added new config var `DATABASE_URL`
- Removed tightly coupled config vars (MONGO_USER, MONGO_PASSWORD, MONGO_DB)
- Introduced additional helper utils

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -17,14 +18,22 @@ const server = new ApolloServer({
 
 app.use(cors());
 server.applyMiddleware({ app, bodyParserConfig: bodyParser.json() });
-const url = `mongodb+srv://${process.env.MONGO_USER}:${process.env.MONGO_PASSWORD}@cluster0-yt8sz.mongodb.net/${process.env.MONGO_DB}`;
-mongoose.connect(url, { useNewUrlParser: true })
+
+const url = env('DATABASE_URL');
```
Comment (Allan690) [inline app.js]: Will need to add this variable to circle CI for the tests to pass

Comment (devdbrandy) [reply app.js]: Yeah, its needed.

Comment (devdbrandy) [reply app.js]: @Allan690, I double checked the information regarding the `DATABASE_URL` and I realise that the test fails on my local machine even when the config var is provided. And also, there's an error `port already in use` as a result of an existing running server.

So I took the liberty of abstracting the server listener, and database connection to a separate file, that way we don't have to listen to a port every-time we run the test.

Comment (devdbrandy) [reply app.js]: Thinking about, are there going to be integration tests for model operation, or a simple mock will do?

Comment (Allan690) [reply app.js]: > @Allan690, I double checked the information regarding the `DATABASE_URL` and I realise that the test fails on my local machine even when the config var is provided. And also, there's an error `port already in use` as a result of an existing running server.
> 
> So I took the liberty of abstracting the server listener, and database connection to a separate file, that way we don't have to listen to a port every-time we run the test.

I like the implementation. Nice work there.

Comment (Allan690) [reply app.js]: > Thinking about, are there going to be integration tests for model operation, or a simple mock will do?

A simple mock could do for now.

Comment (devdbrandy) [reply app.js]: Alright, cool. So this is good to go 🚀

**Pull request merged**